Source/WebCore/Modules/mediastream/gstreamer/GStreamerIceAgent.cpp

```diff
@@ -0,0 +1,783 @@
+{
+    auto backend = WEBKIT_GST_WEBRTC_ICE_BACKEND(ice);
+
+    auto isClosed = backend->priv->agentIsClosed.load();
+    if (isClosed)
+        return;
+
+    bool shouldWait = promise == nullptr;
+    backend->priv->closePromise = adoptGRef(promise);
+    auto now = WTF::MonotonicTime::now().secondsSinceEpoch();
+    rice_agent_close(backend->priv->agent.get(), now.nanoseconds());
+    webkitGstWebRTCIceAgentWakeup(backend);
+
+    isClosed = backend->priv->agentIsClosed.load();
+    if (!shouldWait || isClosed)
+        return;
+
+    while (!isClosed) {
+        g_main_context_iteration(backend->priv->runLoop->mainContext(), FALSE);
+        isClosed = backend->priv->agentIsClosed.load();
+    }
+}
+#endif
+
+static void webkitGstWebRTCIceAgentFinalize(GObject* object)
+{
+    auto backend = WEBKIT_GST_WEBRTC_ICE_BACKEND(object);
+
+    if (backend->priv->onCandidateNotify)
+        backend->priv->onCandidateNotify(backend->priv->onCandidateData);
+    G_OBJECT_CLASS(webkit_gst_webrtc_ice_backend_parent_class)->finalize(object);
+}
+
+static void webkitGstWebRTCIceAgentConstructed(GObject* object)
+{
+    G_OBJECT_CLASS(webkit_gst_webrtc_ice_backend_parent_class)->constructed(object);
+
+    auto backend = WEBKIT_GST_WEBRTC_ICE_BACKEND(object);
+    auto priv = backend->priv;
+
+    priv->agentIsClosed.exchange(false);
+
+    static Atomic<uint32_t> counter = 0;
+    auto id = counter.load();
+    auto threadName = makeString("webrtc-rice-"_s, id);
+    counter.exchangeAdd(1);
+
+    priv->runLoop = RunLoop::create(ASCIILiteral::fromLiteralUnsafe(threadName.ascii().data()));
+    priv->agent = adoptGRef(rice_agent_new(true, true));
+}
+
+static void findStreamAndApply(const Vector<std::unique_ptr<WebKitGstRiceStream>>& streams, unsigned streamId, Function<void(const WebKitGstIceStream*)> callback)
+{
+    auto index = streams.findIf([streamId](const auto& item) {
+        return item->riceStreamId == streamId;
+    });
+    if (index == notFound) [[unlikely]]
+        return;
+    callback(WEBKIT_GST_WEBRTC_ICE_STREAM(streams[index]->stream.get()));
+}
+
+static void webkitGstWebRTCIceAgentConfigure(WebKitGstIceAgent* backend, RefPtr<SocketProvider>&& socketProvider, ScriptExecutionContextIdentifier identifier)
+{
+    auto priv = backend->priv;
+    priv->socketProvider = WTFMove(socketProvider);
+    priv->identifier = identifier;
+    priv->backendClient = RiceBackendClient::create();
+    priv->iceBackend = priv->socketProvider->createRiceBackend(*priv->backendClient);
+    priv->backendClient->setIncomingDataCallback([weakThis = GThreadSafeWeakPtr(backend)](unsigned streamId, RTCIceProtocol protocol, String&& from, String&& to, SharedMemory::Handle&& data) mutable {
+        auto self = weakThis.get();
+        if (!self)
+            return;
+        findStreamAndApply(self->priv->streams, streamId, [protocol, from = WTFMove(from), to = WTFMove(to), data = WTFMove(data)](const auto stream) mutable {
```

Comment:
In all these calls, if not properly naming the type of the lambda parameter, at least do a `const auto*` . Here and in all calls to `findStringAndApply`.